pkg/apis/batch/v1/defaults.go

```diff
@@ -62,6 +62,15 @@ func SetDefaults_Job(obj *batchv1.Job) {
 			}
 		}
 	}
+	if obj.Spec.PodReplacementPolicy == nil {
+		if obj.Spec.PodFailurePolicy != nil {
+			failed := batchv1.Failed
+			obj.Spec.PodReplacementPolicy = &failed
```

Comment:
non-blocking nit: I would suggest to use `podReplacementPolicyPtr` function

Reply:
done.